### ISSUE TITLE
Build Failed Proguard

### ISSUE THREAD

**Issue by pratamawijaya** [VictorAlbertos/RxCache#6]
i have activate proguard
and my project cant be build

i have added 
`-dontwarn io.rx_cache.**` to my proguard-rules.pro but it seems still failed

here are log error from your project [link](https://github.com/VictorAlbertos/RxCacheSamples/tree/master/sample_android) 
i have actived `minifyEnabled:true` and build using `./gradlew clean assembleRelease`
```
Note: there were 9 unresolved dynamic references to classes or interfaces.
      You should check if you need to specify additional program jars.
      (http://proguard.sourceforge.net/manual/troubleshooting.html#dynamicalclass)
Note: there were 11 accesses to class members by means of introspection.
      You should consider explicitly keeping the mentioned class members
      (using '-keep' or '-keepclassmembers').
      (http://proguard.sourceforge.net/manual/troubleshooting.html#dynamicalclassmember)
Warning: there were 1615 unresolved references to classes or interfaces.
         You may need to add missing library jars or update their versions.
         If your code works fine without the missing classes, you can suppress
         the warnings with '-dontwarn' options.
         (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedclass)
Warning: there were 1 unresolved references to library class members.
         You probably need to update the library versions.
         (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedlibraryclassmember)
Warning: Exception while processing task java.io.IOException: Please correct the above warnings first.
:sample_android:transformClassesAndResourcesWithProguardForRelease FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sample_android:transformClassesAndResourcesWithProguardForRelease'.
> java.io.IOException: Please correct the above warnings first.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 8.612 secs
```

have you trying build using proguard activate..? any suggestion proguard config for rxcache ?  thanks

**Comment by VictorAlbertos** [comment]
I don’t use proguard. Although I tried it long time ago, I decided to not use it because you need to know very well what are you doing in order to not get fucked at some unexpected point :)

But in your log trace, lines error don’t point to specific classes, so I can’t tell you much. But I can tell you that RxCache uses internally Gson and Guava, so I guess you will need to configure proguards rules for both of them too.   

Why are you using proguard, anyway? Is it because the [65K Reference Limit](http://developer.android.com/intl/es/tools/building/multidex.html)?

I’m thinking to give the users of RxCache the option to exclude Guava in order to reduce the method count in more than 13.000 lines, of course doing so the library will not be able to evict cached objects automatically before the application reaches its max heap memory limit. But I think most applications don’t require this feature at all; because, due to its mount of data, they will never reach the max heap memory limit.

**Comment by pratamawijaya** [comment]
yes, im using proguard for 64k limit and shrinking my resource, 
may i ask another question,

it seem when i have JSON data like these

```
{
  "stories": [
    {
      "id": "0c57977a-adeb-11e5-832c-7fe3bd1b3c39",
      "content_url": "http://musik.kapanlagi.com/berita/demi-bermusik-petty-x-factor-indonesia-cuti-satu-tahun-dari-sma-9927b6.html",
      "title": "Demi Bermusik, Petty X Factor Indonesia Cuti Satu Tahun Dari SMA",
      "image_url": "http://cdn.klimg.com/kapanlagi.com/p/headline/demi-bermusik-petty-x-factor-indonesia--9927b6.jpg",
      "published_at": 1450002120,
      "faved_at": 1451366074,
      "faver": {
        "id": "8adb74ef-f9ad-47a4-a9c3-ba8c8fbe94cd",
        "username": "nicholas.two"
      }
    }
]}
```

i have created pojo like these
```
public class Faver  {
  public String id;
  public String username;
}
```
and 
```
public class Story {
  @SerializedName("content_url") public String contentUrl;
  public String title;
  @SerializedName("image_url") public String imageUrl;
  public String id;
  @SerializedName("faved_at") public String favedAt;
  @SerializedName("published_at") public String publishedAt;
  @SerializedName("faver") public Faver faver;
}
```

it seem `Faver` object not saved on the disk, any idea what happen..?

**Comment by VictorAlbertos** [comment]
Thanks to you for using it ;)

If you find a proper proguard configuration rules, please share it in order to add it to the README.  

In any case, I think I'm going to adapt the library to allow to no use Guava if the user does not need automatic evicting objects upon limit heap memory threshold.

**Comment by VictorAlbertos** [comment]
I’ve published a [0.4.9 release](https://github.com/VictorAlbertos/RxCache/releases/tag/0.4.9) which supports guava exclusion. 
It’s reflected in the README file.
